### ISSUE TITLE
IDE "Syntactic sugar" - collapsible arbitrary code regions

### ISSUE THREAD

**Issue by mburns08109** [twinbasic/twinbasic#1070]
**Is your feature request related to a problem? Please describe.**
(not sure if this has been created as a spec item as yet, so am submitting it here.)
Not really a problem - unless you count overuse of screen real estate as a problem

**Describe the solution you'd like**
Give us a means akin to the T-SQL /* ...*/ block comment structure (I'm totally open on the actual syntax to be applied here)
such that:
1) the code lines will be able to appear in a collapsed state
2) a "substitute" line can/will be shown in place of the collapsed code lines in the IDE (ONLY) - this is NOT to alter compiled code in any way...only to collapse down the display of multiple code lines into a "visual placeholder" line (with the [+] preceding it to expand the lines fully

So, a suggested syntax might be something like:
`

/+ {controls(MyTxtbx(01...15).value = ...various}
me.MyTxtBx01.value = me.recordset.fields("01").value
me.MyTxtBx02.value = me.recordset.fields("02").value
me.MyTxtBx03.value = nz(me.recordset.fields("03").value, "N/A")
'... skipped for brevity
me.MyTxtBx15.value = me.recordset.fields("15").value
+/

`

which would show in the IDE as:

`

[+] controls(MyTxtbx(01...15).value = ...various

`

until the [+] is clicked, which would expand the inner lines and hide the { text } line.

**Describe alternatives you've considered**
not doing anything and just having some code sections that are annoyingly repetitive and take up tons of screen scroll space

**Additional context**
I would find this useful for those sections where multiple similar lines of code were used to accomplish things similar to the above, but that could be often shrunk down once unit tested to reduce excessive vertical scrolling and to help make indented source code control flows easier to visualize and maintain.

**Comment by wqweto** [comment]
Btw, this is not what a “syntactic sugar” means. There is no “transformation on the level of syntax which the compiler does” in this proposition

**Comment by mburns08109** [comment]
Agreed - that's why I'd labelled it as "_IDE "Syntactic sugar"_ " in quotes because that's not what it really is, and "IDE" as a preface because that's where the _"syntactic sugar"_ would apply (except for any compiler changes needed to ignore the needed things it has to ignore in the source files, of course).

**Comment by mburns08109** [comment]
ping
